### PR TITLE
Remove single moq usage

### PR DESCRIPTION
**Pull request by pakrym**
Issue: https://github.com/aspnet/Hosting/issues/499
@Tratcher @pranavkm

### CODE REVIEW

File: test/Microsoft.AspNet.Hosting.Tests/project.json [hunk 1]
```diff
@@ -9,7 +9,6 @@
         "Microsoft.AspNet.Testing": "1.0.0-*",
         "Microsoft.Extensions.OptionsModel": "1.0.0-*",
         "Microsoft.Extensions.PlatformAbstractions": "1.0.0-*",
-        "Moq": "4.2.1312.1622",
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Hosting.Tests/project.json]: Can you add `dnxcore50` to this project?

Comment (Tratcher) [inline test/Microsoft.AspNet.Hosting.Tests/project.json]: Both test projects, right?

Comment (pakrym) [inline test/Microsoft.AspNet.Hosting.Tests/project.json]: Done, the other one has it https://github.com/aspnet/Hosting/blob/pakrym/remove-moq/test/Microsoft.AspNet.TestHost.Tests/project.json#L17

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**